src/Qir/Controller/Driver/QirDriverGenerator.cs

```diff
@@ -5,22 +5,48 @@
 using System.IO;
 using System.Threading.Tasks;
 using Microsoft.Quantum.Qir.Utility;
+using Microsoft.Quantum.QsCompiler;
 using Microsoft.Quantum.QsCompiler.BondSchemas.EntryPoint;
 
 namespace Microsoft.Quantum.Qir.Driver
 {
     public class QirDriverGenerator : IQirDriverGenerator
     {
+        private const string BytecodeFileName = "qir.bc";
+        private const string DriverFileName = "driver.cpp";
         private readonly ILogger logger;
 
         public QirDriverGenerator(ILogger logger)
         {
             this.logger = logger;
         }
 
-        public Task GenerateQirDriverCppAsync(DirectoryInfo sourceDirectory, EntryPointOperation entryPointOperation, ArraySegment<byte> bytecode)
+        public async Task GenerateQirDriverCppAsync(DirectoryInfo sourceDirectory, EntryPointOperation entryPointOperation, ArraySegment<byte> bytecode)
         {
-            throw new NotImplementedException();
+            // Wrapped in a task because DirectoryInfo operations are not asynchronous.
+            await Task.Run(async () =>
+            {
+                sourceDirectory.Create();
+                logger.LogInfo($"Created source directory at {sourceDirectory.FullName}.");
+
+                // Create driver.
+                var driverFile = new FileInfo(Path.Combine(sourceDirectory.FullName, DriverFileName));
+                using var driverFileStream = driverFile.OpenWrite();
+                GenerateQirDriverCppHelper(entryPointOperation, driverFileStream);
+                logger.LogInfo($"Created driver file at {driverFile.FullName}.");
+
+                // Create bytecode file.
+                var bytecodeFile = new FileInfo(Path.Combine(sourceDirectory.FullName, BytecodeFileName));
+                using var bytecodeFileStream = bytecodeFile.OpenWrite();
+                await bytecodeFileStream.WriteAsync(bytecode.Array, bytecode.Offset, bytecode.Count);
+                logger.LogInfo($"Created bytecode file at {bytecodeFile.FullName}.");
+            });
+        }
+
+        // Virtual method wrapper is to enable mocking in unit tests.
+        public virtual void GenerateQirDriverCppHelper(EntryPointOperation entryPointOperation, Stream stream)
```

Comment:
Since the main purpose of this wrapper is to enable mocking in tests, this could be `internal`.

Reply:
Yeah if this were a library I'd mark it internal, but since it isn't, there is no appreciable difference with that approach (except for the requirement then to add an InternalsVisibleTo)